pkg/api/types.go

```diff
@@ -27,87 +52,152 @@ type BrokerStatus struct {
+	kunversioned.TypeMeta
+	kapi.ObjectMeta
+
+	// BrokerName is the reference to the Broker that provides this service.
+	BrokerName string
 
-	// WATCH THE MAIL FOR MY INVOICE
+	Bindable      bool
+	Plans         []ServicePlan
+	PlanUpdatable bool // Do we support this?
 
-	ID string
+	// Move to annotation
 	Description string
-	Bindable bool
-	PlanUpdatable bool // TODO: do we support this? document if we don't
-	Tags []string
-	Requires []string
-	Metadata interface{}
-	Plans []ServicePlan
-	DashboardClient interface{}
-	BrokerName string // broker object name
+
+	// CF-specific; move to annotation
+	// CFGUID is the identity of this object for use with the CF SB API.
+	// Immutable.
+	CFGUID string
+
+	// CF-specific; move to annotations
+	CFTags                    []string
+	CFRequires                []string
+	CFMaxDBPerNode            string
+	CFMetadata                interface{}
+	CFDashboardOAuth2ClientID string
+	CFDashboardSecret         string
+	CFDashboardRedirectURI    string
 }
 
 type ServicePlan struct {
-	ID string
+	// CLI-friendly name of this plan
 	Name string
+
+	// Move to annotation
 	Description string
-	Metadata interface{}
-	Free bool
-	// OTHER?
+
+	// CF-specific; move to annotation
+	// CFGUID is the identity of this object for use with the CF SB API.
+	// Immutable.
+	CFGUID string
+
+	// CF-specific; move to annotations
+	CFMetadata interface{}
+	CFFree     bool
 }
 
 type Instance struct {
-	// boilerplate
+	kunversioned.TypeMeta
+	kapi.ObjectMeta
 
-	Spec InstanceSpec
-	Status InstanceStatus
+	Spec   InstanceSpec
+	Status InstanceState
 }
 
 type InstanceSpec struct {
-	ServiceClassName string // name of service class resource
+	// ServiceClassName is the reference to the ServiceClass this is an instance of.
+	ServiceClassName string
+	// ServicePlanName is the reference to the ServicePlan for this instance.
 	PlanName string
-	InstanceGUID string // may move this to an annotation
-	Credentials string // this is legacy CF stuff
-	DashboardURL string
-	InternalID string // came from ville. remove? nobody likes that guy
-	CFServiceID string
-	CFPlanID string
-	CFType string // may move this to an annotation
-	CFSpaceGUID string // may move this to an annotation
-	CFLastOperation string // TODO: talk about supporting async provision
-	CFParameters map[string]interface{}
+
+	Parameters map[string]interface{}
+
+	// CF-specific; move to annotation
+	// CFGUID is the identity of this object for use with the CF SB API.
+	// Immutable.
+	CFGUID string
+
+	// CF-specific; move to annotations
+	CFCredentials   string
+	CFDashboardURL  string
+	CFInternalID    string // came from ville. remove? nobody likes that guy
+	CFServiceID     string
+	CFPlanID        string
+	CFType          string
+	CFSpaceGUID     string
+	CFLastOperation string
 }
 
-type InstanceStatus struct {
-	Status string // TODO: make this an "enum" (constant + type)
+type InstanceState struct {
+	Status InstanceState
```

Comment:
this type should be called `InstanceStatus`. right now, it's a circular dependency